src/main/kotlin/com/charleskorn/kaml/YamlInput.kt

```diff
@@ -334,7 +347,7 @@ private class YamlTaggedInput(val taggedNode: YamlTaggedNode, context: SerialMod
         isPolymorphic = kind === UnionKind.POLYMORPHIC
     }
 
-    private inline fun <T> maybeCallOnChild(blockOnTag: () -> T, blockOnChild: YamlInput.() -> T): T {
+    private inline fun <T> maybeCallOnChild(functionName: String = "", blockOnTag: () -> T = { throw IllegalArgumentException("can't call $functionName on tag") }, blockOnChild: YamlInput.() -> T): T {
```

Comment:
Rather than the default values for parameters, could this be rewritten as two overloads? I'm imagining one that takes the two lambdas, and another that takes the function name and `blockOnChild`